lib/cloudregister/registerutils.py

```diff
@@ -509,6 +510,47 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def setup_registry(registry_fqdn, username, password):
+    set_registry_auth_token(registry_fqdn, username, password)
+
+
+# ----------------------------------------------------------------------------
+def set_registry_auth_token(registry_fqdn, username, password):
+    """Set the auth token to pull images from SUSE registry."""
+    auth_token = base64.b64encode('{username}:{password}'.format(
+        username=username,
+        password=password
+    ).encode()).decode()
+    registry_credentials = {}
+    registry_credentials[registry_fqdn] = {'auth': auth_token}
+
+    config_json = {}
+    try:
+        with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+            config_json = json.load(cred_json)
+        # file exists
+        # set the new registry credentials,
+        # independently of what that content was
+        config_json['auths'].update(registry_credentials)
+    except (FileNotFoundError, KeyError):
+        # config file does not exist or "auths" key is not set
+        os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH), exist_ok=True)
```

Comment:
This is rather in-discriminant, if the file doesn't exist we also assume that the directory doesn't exist and then try to create it. If we are going to be in-discriminant about the operation and we know we need the directory why do we not set it up in the wrapper function i.e. in `setup_registry` there we could have 

```
if not os.path.exists(os.path.dirname(REGISTRY_CREDENTIALS_PATH)):
    os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH)
```

The directory is one of the fundamental things for setting up the registry on the client as such it should be at the highest level possible. Then we also do not have to handle 2 different things here.

Reply:
Creating the dir as early as possible does not eliminate the handling of the 2 different things in this case:

- if dir did not exist, and we created it, the file won't exist
- if dir did exist does not guarantee the file to be present anyway

I will check for the dir as early as possible as suggested but handling both exception